agent_build/package_builders.py

```diff
@@ -0,0 +1,802 @@
+            "scalyr_agent.builtin_monitors.graphite_monitor",
+            "scalyr_agent.builtin_monitors.mysql_monitor",
+            "scalyr_agent.builtin_monitors.nginx_monitor",
+            "scalyr_agent.builtin_monitors.shell_monitor",
+            "scalyr_agent.builtin_monitors.syslog_monitor",
+            "scalyr_agent.builtin_monitors.test_monitor",
+            "scalyr_agent.builtin_monitors.url_monitor",
+        ]
+
+        # Add packages to frozen binary paths.
+        paths_to_include = [
+            str(scalyr_agent_package_path),
+            str(scalyr_agent_package_path / "builtin_monitors"),
+        ]
+
+        # Add platform specific things.
+        if platform.system().lower().startswith("linux"):
+            hidden_imports.extend(
+                [
+                    "scalyr_agent.builtin_monitors.linux_system_metrics",
+                    "scalyr_agent.builtin_monitors.linux_process_metrics",
+                ]
+            )
+
+            tcollectors_path = pl.Path(
+                __SOURCE_ROOT__,
+                "scalyr_agent",
+                "third_party",
+                "tcollector",
+                "collectors",
+            )
+            add_data.update(
+                {tcollectors_path: tcollectors_path.relative_to(__SOURCE_ROOT__)}
+            )
+        elif platform.system().lower().startswith("win"):
+            hidden_imports.extend(
+                [
+                    "scalyr_agent.builtin_monitors.windows_event_log_monitor",
+                    "scalyr_agent.builtin_monitors.windows_system_metrics",
+                    "scalyr_agent.builtin_monitors.windows_process_metrics",
+                ]
+            )
+
+        # Create --add-data options from previously added files.
+        add_data_options = []
+        for src, dest in add_data.items():
+            add_data_options.append("--add-data")
+            add_data_options.append(f"{src}{os.path.pathsep}{dest}")
+
+        # Create --hidden-import options from previously created hidden imports list.
+        hidden_import_options = []
+        for h in hidden_imports:
+            hidden_import_options.append("--hidden-import")
+            hidden_import_options.append(str(h))
+
+        dist_path = pyinstaller_output / "dist"
+
+        # Run the PyInstaller.
+        subprocess.check_call(
+            [
+                sys.executable,
+                "-m",
+                "PyInstaller",
+                str(scalyr_agent_package_path / "agent_main.py"),
+                "--onefile",
+                "--distpath",
+                str(dist_path),
+                "--workpath",
+                str(pyinstaller_output / "build"),
+                "-n",
+                type(self).FROZEN_BINARY_FILE_NAME,
+                "--paths",
+                ":".join(paths_to_include),
+                *add_data_options,
+                *hidden_import_options,
+                "--exclude-module",
```

Comment:
Any reason why we exclude those modules? Would be good to add a comment and clarify why we need to exclude those.

Reply:
Correct, that's all about the size, I'll add the commant.